x-pack/filebeat/input/httpjson/encoding.go

```diff
@@ -176,3 +180,33 @@ func decodeAsCSV(p []byte, dst *response) error {
 
 	return nil
 }
+
+func decodeAsZip(p []byte, dst *response) error {
+	var results []interface{}
+	r, err := zip.NewReader(bytes.NewReader(p), int64(len(p)))
+	if err != nil {
+		return err
+	}
+
+	for _, f := range r.File {
+		rc, err := f.Open()
+		if err != nil {
+			return err
+		}
+
+		dec := json.NewDecoder(rc)
+		for dec.More() {
+			var o interface{}
+			if err := dec.Decode(&o); err != nil {
+				func() { _ = rc.Close() }()
```

Comment:
What's the reason for the function wrapper? Why not just `rc.Close()`?

Reply:
Local linting, removed